src/components/Fieldset.tsx

```diff
@@ -0,0 +1,98 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+
+import "lbh-frontend/lbh/components/lbh-fieldset/_fieldset.scss";
+
+/**
+ * The proptypes for {@link FieldsetLegend}.
+ */
+export interface FieldsetLegendProps {
+  id?: string;
+  className?: string;
+  isPageHeading?: boolean;
+  children?: React.ReactNode;
+}
+
+/**
+ * A component to set a legend for a {@link Fieldset} by passing a {@link FieldsetProps.legend} prop.
+ *
+ * ```ts
+ * // Creating a fieldset with a legend
+ * <Fieldset
+ *   id="fieldsetId"
+ *   className="fieldsetClass"
+ *   describedBy="fieldsetDescribedBy"
+ *   role="fieldsetRole"
+ *   legend={
+ *     <FieldsetLegend id="legendId" className="legendClass" isPageHeading>
+ *       children
+ *     </FieldsetLegend>
+ *   }
+ * >
+ *    children
+ * </Fieldset>
+ * ```
+ */
+export const FieldsetLegend: React.FunctionComponent<FieldsetLegendProps> = (
```

Comment:
Let's do the same thing as with buttons and pull these into separate files in their own folder. Feels a bit messy to have both components in here.